core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -246,7 +247,8 @@ public void onNext(HttpObject o) {
                 if (o instanceof HttpHeaders) {
                     final HttpHeaders trailers = (HttpHeaders) o;
                     if (trailers.contains(HttpHeaderNames.STATUS)) {
-                        throw newIllegalStateException("published a trailers with status: " + o);
+                        fail(new IllegalArgumentException("published a trailers with status: " + o));
```

Comment:
I guess `IllegalStateException` is better in this class because the exception does not go to the caller.

Reply:
Never mind. It's good as it is.

Reply:
Had a chat with @trustin and I slightly prefer to leave this as it is because the user will see this exception from `RequestLog` even though the exception does not go to the caller right away.
But I'm not so strong on this. 😆 Any idea? @anuraaga, @ikhoon

Reply:
Okay then, let me just use `IllegalArgumentException`. 😉